pkg/startup/startup.go

```diff
@@ -244,6 +222,61 @@ func Run() {
 		log.WithError(err).Errorf("Unable to ensure network for os")
 		terminate()
 	}
+
+	if !runOnce {
+		autoDetectPollingInterval := DEFAULT_AUTODETECT_POLL_INTERVAL
+		if os.Getenv("AUTODETECT_POLL_INTERVAL") != "" {
+			autoDetectPollingInterval, _ = time.ParseDuration(os.Getenv("AUTODETECT_POLL_INTERVAL"))
+		}
+		for {
+			select {
+			case <-time.After(autoDetectPollingInterval):
+				checkConflicts = setupIPAndSubnets(ctx, node, cli)
+				if checkConflicts {
+					checkNodeIPConflictCheck(ctx, node, cli)
+					// Apply the updated node resource.
+					if _, err := CreateOrUpdate(ctx, cli, node); err != nil {
```

Comment:
We need to do this conditionally - namely, we don't want to update the node object if it hasn't changed, so we should compare to see if we actually need to make the API request.

Reply:
added a separate function to address this comment.